tools/dds/dds-sniffer/dds-sniffer.cpp

```diff
@@ -354,35 +360,31 @@ void dds_sniffer::save_topic_reader( const eprosima::fastrtps::rtps::ReaderDisco
 {
     std::string topic_name = info.info.topicName().c_str();
     _discovered_topics[topic_name].readers.insert( info.info.guid() );
-    save_max_indentation( std::move( topic_name ) );
 }
 
 void dds_sniffer::remove_topic_reader( const eprosima::fastrtps::rtps::ReaderDiscoveryInfo & info )
 {
-    auto topic_entry = _discovered_topics[info.info.topicName().c_str()];
-    auto reader_iter = topic_entry.readers.find( info.info.guid() );
-    if( reader_iter != topic_entry.readers.end() )
+    auto topic_entry = _discovered_topics.find(info.info.topicName().c_str());
+    if(topic_entry != _discovered_topics.end() )
     {
-        topic_entry.readers.erase( reader_iter );
-        if( topic_entry.writers.empty() && topic_entry.readers.empty() )
+        topic_entry->second.readers.erase( info.info.guid() );
+        if( topic_entry->second.writers.empty() && topic_entry->second.readers.empty() )
         {
             _discovered_topics.erase( info.info.topicName().c_str() );
         }
     }
 }
 
-void dds_sniffer::save_max_indentation( const std::string&& str )
+void dds_sniffer::calc_max_indentation()
 {
-    std::istringstream topic( str );
-    std::string nested;
     uint32_t indentation = 0;
-    while (std::getline( topic, nested, '/' ))  // Use / as delimiter for nested topic names
+    for (auto topic : _discovered_topics)
     {
-        ++indentation;
+        indentation = static_cast<uint32_t>(std::count( topic.first.begin(), topic.first.end(), '/' ));  // Use / as delimiter for nested topic names
     }
-    if (indentation > _max_indentation)
+    if (indentation >= _max_indentation)
```

Comment:
Why only increase it?
If we remove some writers, there should be a new indent, even if it's less than before, no?

Reply:
We use this function only at print time.
Previously we tracked additions and subtractions of entities, but it was wasteful. Now we calculate the indentation for pretty printing only when, and if, needed